masu/util/ocp/common.py

```diff
@@ -131,14 +126,35 @@ def get_cluster_id_from_provider(provider_uuid, schema):
         LOG.warning(err_msg)
         return cluster_id
 
-    with OCPReportDBAccessor(schema, column_map) as report_accessor:
-        usage_period_qry = report_accessor.get_usage_period_query_by_provider(provider_obj.id)
-        clusters = []
-        for obj in usage_period_qry.all():
-            clusters.append(obj.cluster_id)
-        if len(set(clusters)) > 1:
-            err_msg = 'Provider {} is associated with too many clusters. Clusters: {}'.\
-                format(provider_uuid, str(set(clusters)))
-            raise MasuConfigurationError(err_msg)
-        cluster_id = clusters.pop() if clusters else None
+    with ProviderDBAccessor(provider_uuid=provider_uuid) as provider_accessor:
+        cluster_id = provider_accessor.get_authentication()
+
     return cluster_id
+
+
+def get_provider_uuid_from_cluster_id(cluster_id):
+    """
+    Return the provider UUID given the cluster ID.
+
+    Args:
+        cluster_id (String): OpenShift Cluster ID
+
+    Returns:
+        (String): provider UUID
+
+    """
+    provider_uuid = None
+    auth_id = None
+    with ProviderAuthDBAccessor(provider_resource_name=cluster_id) as auth_accessor:
+        auth_id = auth_accessor.get_auth_id()
+        if auth_id:
+            with ProviderDBAccessor(auth_id=auth_id) as provider_accessor:
+                provider_uuid = provider_accessor.get_uuid()
+    return provider_uuid
+
+
+def poll_ingest_override_for_provider(provider_uuid):
+    """Determine if provider should be ingested via the poller."""
```

Comment:
Can you expand this docstring to be more descriptive, please?

It's not clear to me what the function's purpose is and how it fits into the process.

What is it overriding and why does that need to be overridden? Looking at the code, it doesn't seem to be overriding anything, but instead validating that a path exists.

Reply:
Yes; I'll update the doc string.  This is to support OCP (any LISTENER based provider) data side loading where you can short cut the upload service for development purposes.